openmdao/core/component.py

```diff
@@ -25,6 +27,10 @@
                       'exact': None}
 
 
+# regex to check for valid names.
+namecheck_rgx = re.compile('[_a-zA-Z0-9\:]+')
```

Comment:
This will allow variable names that are just a simple number or just a colon.  I think we should stick to regular pythonic names plus a possible colon.